tests/std/tests/P0896R4_common_iterator/test.cpp

```diff
@@ -74,28 +118,85 @@ struct instantiator {
+                    assert(diff_it_it == -1);
+
+                    // Difference iterator / sentinel
+                    const same_as<iter_difference_t<Iter>> auto diff_it_sen = Cit{Iter{input + 1}} - Cit{Sen{input}};
+                    const same_as<iter_difference_t<Iter>> auto diff_sen_it = Cit{Sen{input}} - Cit{Iter{input + 1}};
+                    assert(diff_it_sen == 1);
+                    assert(diff_sen_it == -1);
+
+                    // Difference sentinel / sentinel
+                    const same_as<iter_difference_t<Iter>> auto diff_sen_sen = Cit{Sen{input}} - Cit{Sen{input + 1}};
+                    assert(diff_sen_sen == 0);
+
+                    // Difference iterator / other iterator
+                    const same_as<iter_difference_t<Iter>> auto diff_it_oit = Cit{Iter{input}} - OCit{Iter{input + 1}};
+                    assert(diff_it_oit == -1);
+
+                    // Difference iterator / other sentinel
+                    const same_as<iter_difference_t<Iter>> auto diff_it_osen = Cit{Iter{input + 1}} - OCit{OSen{input}};
+                    assert(diff_it_osen == 1);
+
+                    // Difference other iterator / sentinel
+                    const same_as<iter_difference_t<Iter>> auto diff_sen_oit = Cit{Sen{input}} - OCit{Iter{input + 1}};
+                    assert(diff_sen_oit == -1);
+
+                    // Difference sentinel / other sentinel
+                    const same_as<iter_difference_t<Iter>> auto diff_sen_osen = Cit{Sen{input}} - OCit{OSen{input + 1}};
+                    assert(diff_sen_osen == 0);
+                }
             }
 
             { // [common.iter.cust]
                 if constexpr (input_iterator<Iter>) { // iter_move
                     Cit iter1{Iter{input}};
 
-                    const auto iter2 = ranges::iter_move(iter1);
-                    assert(iter2 == 1);
+                    const same_as<iter_value_t<Iter>> auto element1 = ranges::iter_move(iter1);
+                    assert(element1 == P(0, 1));
                 }
                 if constexpr (indirectly_swappable<Iter>) { // iter_swap
                     Cit iter1{Iter{input}};
                     Cit iter2{Iter{input + 1}};
 
                     ranges::iter_swap(iter1, iter2);
-                    assert(*iter1 == 2);
-                    assert(*iter2 == 1);
+                    assert(*iter1 == P(0, 2));
+                    assert(*iter2 == P(0, 1));
                 }
             }
         }
     }
 };
 
+bool test_operator_arrow() {
+    P input[3] = {{0, 1}, {0, 2}, {0, 3}};
+
+    using pointerTest = common_iterator<P*, void*>;
+    pointerTest pointerIter{input};
+
+    assert(*pointerIter == P(0, 1));
+    assert(pointerIter->first == 0);
+    assert(pointerIter->second == 1);
+    static_assert(is_same_v<remove_cvref_t<decltype(pointerIter.operator->())>, P*>);
+    static_assert(!is_same_v<decltype(pointerIter.operator->()), P*>);

```

Comment:
Why not simply `static_assert(is_same_v<decltype(pointerIter.operator->()), P* const&>)`?